src/handwritten/task.py

```diff
@@ -79,15 +80,15 @@ def __init__(self, new_task_name='', *, grpc_options=None):
             raise DaqError(
                 f'Unsupported session name: "{grpc_options.session_name}". If a session name is specified, it must match the task name.',
                 DAQmxErrors.UNKNOWN,
-                task_name=self.name)
+                task_name=new_task_name)
         
         self._interpreter = utils._select_interpreter(grpc_options)
         self._handle, self._close_on_exit = self._interpreter.create_task(new_task_name)
 
         self._initialize(self._handle, self._interpreter)
 
     def __del__(self):
-        if self._handle:
+        if self._handle is not None and self._close_on_exit:
             warnings.warn(
                 'Task of name "{}" was not explicitly closed before it was '
```

Comment:
I think we need to revisit this for the explicit gRPC detach use case. I filed AB#2383312 about it.